tests/rptest/tests/rack_aware_replica_placement_test.py

```diff
@@ -114,6 +114,13 @@ def test_replica_placement(self, rack_layout_str, num_partitions,
         @param replication_factor defines recplication factor of all partitions.
         """
 
+        # https://github.com/redpanda-data/redpanda/issues/11276
+        if self.debug_mode and num_partitions == 400:
+            self.logger.info(
+                "Disabling test in debug mode due to slowness/timeouts with large number of partitions."
+            )
+            return
```

Comment:
this requires `cleanup_on_early_exit` IIRC

Reply:
Done.